presto-docs/src/main/sphinx/connector/elasticsearch.rst

```diff
@@ -0,0 +1,137 @@
+
+For each table defined here, a table description file (see below) must exist.
+
+This property is required; there is no default and at least one table must be defined.
+
+``elasticsearch.default-schema``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Defines the schema which will contain all tables that were defined without
+a qualifying schema name.
+
+This property is optional; the default is ``default``.
+
+``elasticsearch.table-description-dir``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+References a folder within Presto deployment that holds one or more JSON
+files (must end with ``.json``) which contain table description files.
+
+This property is optional; the default is ``etc/elasticsearch``.
+
+``elasticsearch.scroll-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When Presto connects to Elasticsearch using an Elasticsearch client, the client will
+use scroll and fetch to get data from Elasticsearch. This property defines the maximum
+number of hits to be returned with each batch of Elasticsearch scroll.
+
+This property is optional; the default is ``1000``.
+
+``elasticsearch.scroll-time``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When Presto connects to Elasticsearch using an Elasticsearch client, the client will
+use scroll and fetch to get data from Elasticsearch. This property defines number of
+milliseconds the Elasticsearch client would keep live for a search context
+
+This property is optional; the default is ``60000``.
+
+Table Definition Files
+----------------------
+
+Elasticsearch maintains documentations and indexes in a highly scalable way. It provides
+full-text search and analytics functionalities. For Presto, Elasticsearch data must be
+mapped into tables and columns to allow queries against the data.
+
+A table definition file consists of a JSON definition for a table. The
+name of the file can be arbitrary but must end in ``.json``.
+
+.. code-block:: none
+
+    {
+        "tableName": ...,
+        "schemaName": ...,
+        "hostAddress": ...,
+        "port": ...,
+        "clusterName": ...,
+        "index": ...,
+        "type": ...
+    }
+
+=============== ========= ============== =============================
+Field           Required  Type           Description
+=============== ========= ============== =============================
+``tableName``   required  string         Presto table name defined by this file.
+``schemaName``  optional  string         Schema which will contain the table. If omitted, the default schema name is used.
+``hostAddress`` required  string         Elasticsearch Search Node host address.
+``port``        required  string         Elasticsearch Search Node port number.
+``clusterName`` required  string         Elasticsearch cluster name.
+``index``       required  string         Elasticsearch index that is backing this table.
+``type``        required  string         Elasticsearch type, which is the class of object that the document represents.
```

Comment:
`class of object` sounds weird, is this a Java class? If so, we can say `is the Java class that the ...`